tutorial/decorators.md

```diff
@@ -0,0 +1,243 @@
+# Decorators
+
+Because Pyccel is converting a dynamically typed language (Python) to statically types ones, It imposes some *restrictions* on the code. One of them is using decorators that have certain options in order to specify how the functions or the variables should be treated in the conversion, Here are the available decorators.
+
+## Stack array
+
+This decorator indicates that all arrays mentioned as argements (of the decorator) should be stored
+on the stack.
+
+This example shows how the decorators can affect the conversion of the array between the supported languages, Pyccel here is told by the decorator `stack array` to store the array `arrat_in_stack` in the stack, For the array `array_in_heap` Pyccel is assuming that it should be stored in the heap:
+
+```python
+from pyccel.decorators import stack_array
+
+@stack_array('array_in_stack')
+def fun1():
+
+     #/////////////////////////
+     #array stored in the stack
+     #////////////////////////
+     array_in_stack = [1,2,3]
+     #////////////////////////
+     #array stored in the heap
+     #////////////////////////
+     array_in_heap = [1,2,3]
```

Comment:
I think we should store an array here rather than a list given that we don't really support lists (`np.array([1,2,3])`)